samples/simple_embedding/CMakeLists.txt

```diff
@@ -52,8 +52,18 @@ iree_native_test(
 
 endif()
 
+iree_compiler_targeting_iree_arch(_IREE_COMPILER_TARGETING_ARM_32 "arm_32")
+iree_compiler_targeting_iree_arch(_IREE_COMPILER_TARGETING_ARM_64 "arm_64")
+iree_compiler_targeting_iree_arch(_IREE_COMPILER_TARGETING_RISCV_32 "riscv_32")
+iree_compiler_targeting_iree_arch(_IREE_COMPILER_TARGETING_RISCV_64 "riscv_64")
+iree_compiler_targeting_iree_arch(_IREE_COMPILER_TARGETING_X86_64 "x86_64")
 if(IREE_HAL_DRIVER_LOCAL_SYNC AND IREE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF AND
-   (IREE_TARGET_BACKEND_LLVM_CPU OR IREE_HOST_BIN_DIR))
+   ((IREE_TARGET_BACKEND_LLVM_CPU AND
+    _IREE_COMPILER_TARGETING_ARM_32 AND
+    _IREE_COMPILER_TARGETING_ARM_64 AND
+    _IREE_COMPILER_TARGETING_RISCV_32 AND
+    _IREE_COMPILER_TARGETING_RISCV_64 AND
+    _IREE_COMPILER_TARGETING_X86_64) OR IREE_HOST_BIN_DIR))
```

Comment:
Oh no... "simple_embedding" is getting even more complex x_x

Reply:
The general issue with `samples/simple_embedding/` is that the `samples/` folder is first and foremost intended to be a source that people can copy code from and adapt to their own uses. Every branch in the CMake file and every IREE-specific CMake variable makes that harder to do. Top level prerequisites for samples are fine, but many of these are buried deep in the file.

Fixing that is out of scope for this PR though.

Reply:
Right, I understand. The tension is between that, and samples being part of what we test on CI, in a bunch of configurations, so that it's generally in a working state. Given that, I think we need nontrivial build logic, but we could hide it behind helpers, so it doesn't make the CMakeLists so intimidating and the target audience could judge how they can adapt it.

Reply:
We could have the samples as its own CMake project. Then we would either include it in the main IREE cmake or just test it separately. I think it would be OK if the samples didn't have great coverage. It shouldn't be providing coverage for IREE itself and for its own purposes it seems reasonable to just make it a super simple thing. If a sample is providing coverage for "this is how a user might use IREE" then perhaps we should split that out into a separate testing directory where we can have build system logic as complex as we want to ensure all configurations work and such.